blockbase/inc/wp-customize-color-palettes.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+require_once 'wp-customize-palette-control.php';
+
+class GlobalStylesColorPalettes {
+	private $palettes = array ();
+
+	function __construct() {
+		add_action( 'customize_register', array( $this, 'build_palettes' ) );
+		add_action( 'customize_register', array( $this, 'color_palette_control' ) );
+		add_action( 'customize_controls_enqueue_scripts', array( $this, 'customize_preview_js' ) );
+	}
+
+	function customize_preview_js() {
+		wp_enqueue_script( 'customizer-color-palettes', get_template_directory_uri() . '/inc/color-palettes-controls.js', array( 'customize-controls' ) );
+		wp_localize_script( 'customizer-color-palettes', 'colorPalettes', $this->palettes );
+	}
+
+	function build_palettes() {
+		$theme_json = WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_raw_data();
+		$default_palette = $theme_json['settings']['color']['palette']['theme'];
+
+		$default_palette_setting = [];
+		foreach( $default_palette as $default_color ) {
+			$default_palette_setting[ $default_color['slug'] ] = $default_color['color'];
+		}
+
+		$this->palettes['default-palette'] = array(
+			'label' => 'Default',
+			'colors' => $default_palette_setting,
+		);
+
+		$custom_palettes = $theme_json['settings']['custom']['colorPalettes'];
```

Comment:
I'm getting an error here w/ Quadrat due to there being no palette's currently defined in theme.json
Likely the Quadrat palette options weren't included in the change (yet).
But seems reasonable that a BB child might not ship with any customizations so probably a relevant use-case.

Reply:
yeah, that fixed disappeared on a conflict, let me get it back

Reply:
almost...
```
Notice: Undefined index: colorPalettes in /var/www/html/wp-content/themes/themes/blockbase/inc/wp-customize-color-palettes.php on line 33
```

Reply:
should be fine now

Reply:
Pushed [this small change](https://github.com/Automattic/themes/pull/4098/commits/1d411e3238b6b519335ca04ec9229deb17077ea5) to only render these controls if there are alternatives to default to show.  Also addressed that error that seemed to stick around despite the recent change.